process/sync/metablock.go

```diff
@@ -157,7 +170,43 @@ func (boot *MetaBootstrap) setLastEpochStartRound() {
 // These methods will execute the block and its transactions. Finally if everything works, the block will be committed
 // in the blockchain, and all this mechanism will be reiterated for the next block.
 func (boot *MetaBootstrap) SyncBlock() error {
-	return boot.syncBlock()
+	err := boot.syncBlock()
+	isErrGetNodeFromDB := err != nil && strings.Contains(err.Error(), core.GetNodeFromDBErrorString)
```

Comment:
extract L174-L180 to a function `syncAccountsIfNeeded(err error)`

Reply:
extraction does not work here. it complicates things. needs internal interface and all.